app/ante/ante.go

```diff
@@ -126,3 +127,33 @@ func (isd IncrementSequenceDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, sim
 
 	return next(ctx, tx, simulate)
 }
+
+type AccountSetupDecorator struct {
+	ak auth.AccountKeeper
+}
+
+func NewAccountSetupDecorator(ak auth.AccountKeeper) AccountSetupDecorator {
+	return AccountSetupDecorator{
+		ak: ak,
+	}
+}
+
+func (asd AccountSetupDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
```

Comment:
I have a couple of questions that I believe should be documented with comments:

- why is this ante handler decorator only for `MsgEthermint` and not `MsgEthereumTx` too?
- any reason why is the decorator on the second position of the chain of ante decorators?

Reply:
`MsgEthereumTx` sets up the account in the `NewAccountVerificationDecorator` so it didn't need to be added separately, `MsgEthermint` uses the `auth` ante handlers which can't be modified, so I added this one to set up the account before the other handlers

I put it as the second one since I figured it would be good to set up the account before the rest of the decorators run, but it could be put right before whichever one validates the account